File: Kevin-Muthiani/Arrays_and_Strings/remove_element.py
# https://leetcode.com/problems/remove-element/
# Given an array nums and a value val, remove all instances of that value 
# in-place and return the new length. Do not allocate extra space for another 
# array, you must do this by modifying the input array in-place with O(1) 
# extra memory. The order of elements can be changed. It doesn't matter what 
# you leave beyond the new length.

def removeElement(nums, val):
    n = 0

    for i in range(len(nums)):
        if nums[i] != val:
            nums[n] = nums[i]
            n += 1
        
    return n


# Test function
if __name__ == "__main__":
    myArray = [3, 2, 2, 3]
    value = 3
    result = removeElement(myArray, value)
    print(result)
    print(myArray)

File: golclinics/Arrays and Strings/remove_element.java
/*Given an integer array nums and an integer val, remove all occurrences of val in nums in-place.
 The relative order of the elements may be changed.

Since it is impossible to change the length of the array in some languages, you must instead have 
the result be placed in the first part of the array nums. More formally, if there are k elements 
after removing the duplicates, then the first k elements of nums should hold the final result. 
It does not matter what you leave beyond the first k*/


public class remove_element {
    public int removeElement(int[] nums, int val) {
       
        int actual_size=0;
        for(int i=0;i<nums.length;i++){
            if (nums[i]!=val){
                nums[actual_size]=nums[i];
                actual_size++;
            }
        }return actual_size;
        
    }
}
